src/js/components/Menu/Menu.js

```diff
@@ -69,57 +69,72 @@ const Menu = props => {
   const MenuIcon = theme.menu.icons.down;
   const iconColor = normalizeColor('control', theme);
   const align = dropProps.align || dropAlign;
+  let controlButtonIndex;
+  if (align.top === 'top') {
+    controlButtonIndex = -1;
+  } else if (align.bottom === 'bottom') {
+    controlButtonIndex = items.length;
+  } else {
+    controlButtonIndex = undefined;
+  }
   const buttonRefs = {};
+  const constants = {
+    none: 'none',
+    tab: 9,
+    menuControlIncludedTop: align.top === 'top' || undefined,
+    menuControlIncludedBottom: align.bottom === 'bottom' || undefined,
+    controlButtonIndex,
+  };
 
-  // to track if focus is within the menu items, see menuOnFocus/menuOnBlur
-  let timeoutID;
-
-  const [activeItemIndex, setActiveItemIndex] = useState(-1);
+  const [activeItemIndex, setActiveItemIndex] = useState(constants.none);
   const [isOpen, setOpen] = useState(open || false);
-  const [isManagingFocus, setManagingFocus] = useState(false);
-
-  useEffect(() => {
-    if (isOpen) {
-      setManagingFocus(true);
-      if (isManagingFocus) {
-        if (align.top === 'bottom') {
-          buttonRefs[0].focus();
-          setActiveItemIndex(0);
-        } else {
-          buttonRefs[items.length].focus();
-          setActiveItemIndex(-1);
-        }
-      }
-    }
-  }, [isManagingFocus]);
 
   const onDropClose = () => {
-    setActiveItemIndex(-1);
+    setActiveItemIndex(constants.none);
     setOpen(false);
-    setManagingFocus(false);
   };
 
   const onDropOpen = () => {
     setOpen(true);
-    setManagingFocus(true);
   };
 
   const onSelectMenuItem = event => {
-    if (activeItemIndex >= 0) {
-      event.preventDefault();
-      event.stopPropagation();
-      buttonRefs[activeItemIndex].click();
+    if (isOpen) {
+      if (activeItemIndex >= 0) {
+        event.preventDefault();
+        event.stopPropagation();
+        buttonRefs[activeItemIndex].click();
+      }
+    } else {
+      onDropOpen();
     }
   };
 
   const onNextMenuItem = event => {
     event.preventDefault();
     if (!isOpen) {
       onDropOpen();
+    } else if (
+      (event.keyCode === constants.tab || event.which === constants.tab) &&
+      ((!constants.menuControlIncludedBottom &&
+        activeItemIndex === items.length - 1) ||
+        (constants.menuControlIncludedBottom &&
```

Comment:
Can we think of a shorter form for this name? maybe bottomMenu? we can add a more elaborating comment where it is defined.